anndata/_core/anndata.py

```diff
@@ -1701,6 +1711,11 @@ def concatenate(
                 for key in shared_layers:
                     layers[key].append(ad[:, vars_intersect].layers[key])
 
+            # obsm
+            if join == "inner":
```

Comment:
Great, but I don't think it should be restricted to merely "inner" as this only relates to the variables. I'll fix that.